source/MainWindow.cpp

```diff
@@ -386,10 +393,21 @@ MainWindow::MessageReceived(BMessage* msg)
 			break;
 
 		case MSG_MERGEWITH:
-			PRINT(("[MSG_MERGEWITH]: Not yet implemented."));
-			// TODO: Implement.
-			// "Merge from..." might be a better idea actually.
+			fMergePanel->Show();
+			break;
+
+		case MSG_MERGE_OPEN_DONE:
+		{
+			//load the selected file
+			entry_ref mergeRef;
+			while (fMergePanel->GetNextSelectedRef(&mergeRef) == B_OK) {
+				fAssocEntry = new BEntry(&mergeRef, true);
+				_Load();
+				fUnsavedChanges = true;
+				fAssocEntry = NULL;
```

Comment:
This should be outside the `while` block as I suggested. You need to make sure that setting it to `NULL` is the right thing to do. As my comments in the suggested code explained, you might have to save `fAssocEntry` before the loop and restore it afterwards.

Reply:
Since it gets used for saving the file, I will save it before the loop and restore it afterwards.